core/identity/src/cli/identity.rs

```diff
@@ -248,30 +225,16 @@ impl IdentityCommand {
                 CommandOutput::object(
                     bus::service(identity::BUS_ID)
                         .send(identity::Unlock::with_id(node_id, password))
-                        .await
-                        .map_err(|e| anyhow::Error::msg(e))?,
+                        .await?,
                 )
             }
             IdentityCommand::Drop { node_or_alias } => {
-                let command: identity::Get = node_or_alias.clone().into();
-                let id = bus::service(identity::BUS_ID)
-                    .send(command)
-                    .await
-                    .map_err(|e| anyhow::Error::msg(e))?;
-                let id = match id {
-                    Ok(Some(v)) => v,
-                    Err(e) => return CommandOutput::object(Err::<(), _>(e)),
-                    Ok(None) => anyhow::bail!("identity not found"),
-                };
+                let id = node_or_alias.resolve().await?;
 
-                if id.is_default {
-                    anyhow::bail!("default identity")
-                }
                 CommandOutput::object(
                     bus::service(identity::BUS_ID)
```

Comment:
TODO: add here a warning that if this identity was used as a default identity it might have related entries in service databases (eg. in `payments.db`).